### PR TITLE
add copyright and license info to file headers

### PR DESCRIPTION
**Pull request by pseyfert**
I did not post contact information to the authors file, we could re-use those from the pad, or those from the git commits. Not sure we need to / want to.

### CODE REVIEW

File: bin/get_fulltext.py [hunk 1]
```diff
@@ -2,6 +2,23 @@
 
 """get_fulltext CLI command"""
```
Comment (mfherbst) [inline bin/get_fulltext.py:4]: Is there a reason, why you do not put the copyright notice below the docstring for the module?

File: AUTHORS.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# The libfulltext authors
```
Comment (mfherbst) [inline AUTHORS.md:1]: Maybe link the `AUTHORS.md` from the `README.md` ?

**Comment by andrenarchy** [comment]
Do we really want to add the copyright+license information to each and every file? In my opinion it is too much redundant information.

File: AUTHORS.md (hunk 2 above)
Comment (pseyfert) [reply AUTHORS.md:1]: agree

**Comment by pseyfert** [comment]
@andrenarchy to be really correct, unfortunately yes. (e.g. https://www.gnu.org/licenses/gpl-howto.en.html) we had a short chat offline if we use the official short version of GPL or a one-line disclaimer (many projects do that), but then we needlessly go onto thinner ice where we have no expertise.

File: bin/get_fulltext.py (hunk 1 above)
Comment (pseyfert) [reply bin/get_fulltext.py:4]: discussed offline that we can keep a single line docstring above the copyright, though each way around has its pros and cons

**Comment by andrenarchy** [comment]
[This answer](https://softwareengineering.stackexchange.com/a/125841) sums my position up nicely. Also: not even the Python standard library does it, e.g. see [here](https://github.com/python/cpython/blob/master/Lib/http/client.py). It's really adding a lot of overhead to the development process and kind of discourages to use a lot of small module files.

**Comment by mfherbst** [comment]
But then on the other hand, with the appropriate tooling (e.g. [vim-templates](https://github.com/tibabit/vim-templates)) this really is not a lot of extra effort, but certainly extra clobber. I am ready to prepare the templates as I am using this plugin for other projects.

So the overhead for development argument I would not really count, but certainly it adds clobber and might yield files which have 90% licence comments and only 10% code. Certainly annoying. 

Overall I would still vote for the "better save than sorry" option and have the long headers at the moment.

@michaelavoigt Do you have further info or pointers regards the legal status?

**Comment by andrenarchy** [comment]
I just think there's no "better safe than sorry" here. The code is covered by copyright automatically and we *grant* others a licence (without the licence people can't do anything with it). And it's really customary to add license information as metadata instead of placing it within the content that is licensed.

**Comment by mfherbst** [comment]
If you are sure then I don't mind not having the headers. I still would vote for the `AUTHORS.md` though, because I think it's nice to distribute the name of the guys, who contributed with the code somewhere.

**Comment by andrenarchy** [comment]
+1 for `AUTHORS.md`! :)

**Comment by pseyfert** [comment]
Hm, I don't want to block but I'm a little bit concerned as my employer's legal dept. was very keen on the "every file" part of their guidelines (2017). I just stripped it down to one single line, do you still see this as problematic?

(Not meant to be final atm, like I didn't update the years and now we could reevaluate the docstring/copyright line order)

Also, comments on the reference of `AUTHORS.md` in the current `README.md`?

File: README.md [hunk 3]
```diff
@@ -36,6 +36,11 @@ Some information, ideas and resources are not yet transfered
 related to this project you can find on the
 [34c3 open science workshop][pad] pad.
 
+# Copyright and License
+
+Copyright (C) 2017-2018 [the libfulltext authors](AUTHORS.md).
```
Comment (mfherbst) [inline README.md]: I would not have the "the" be part of the link but other than that :+1:.

File: setup.py [hunk 4]
```diff
@@ -1,5 +1,8 @@
 """Setup for libfulltext"""
 
+# copyright © 2017 the libfulltext authors (see AUTHORS.md and LICENSE)
+
```
Comment (mfherbst) [inline setup.py]: Nit pick: The two empty lines seem a bit funny.

**Review by mfherbst — DISMISSED** [review]
As I said regarding the order, I'd prefer the licence note first and the docstrings thereafter, but I would not object if we agree to do it otherwise in the end.

Generally I approve the short licence version, but I am still unsure whether this is the thing we should stick to in the end. It definitely makes our intentions clear and thus we should be safe ... unless there is a legal loop hole we are all unaware.

**Review by mfherbst — DISMISSED** [review]
I approve, but please still note my previous review comment regards the order of docstring/licence and legal loopholes.

**Review by andrenarchy — CHANGES_REQUESTED** [review]
I'd also say it's nicer to have the copyright/license hint in the first line. I'm OK with this for now but I'm sure that these comments will be missing in some files soon... Can we add this to our linting process somehow?

Can you update the years to `2017-2018` everywhere directly? Then we don't need another happy new year PR. ;)

The `AUTHORS.md` file looks good!

**Comment by mfherbst** [comment]
We talked to some people ad 34c3 and it seems enough to write the year where the file was first touched, so just `2017` in that philosophy is good enough.

**Comment by andrenarchy** [comment]
@mfherbst fair enough – you know what I think about this anyway! ;)

**Review by mfherbst — APPROVED** [review]

**Review by andrenarchy — APPROVED** [review]

**Pull request merged**